### PR TITLE
Further authorization fixes ...

### PR DESCRIPTION
**Pull request by kevinmtrowbridge**
Hi @AndrewLngdn: 

Can you review this when you get a chance and maybe look around for any other places that might be broken by the authorization refactoring we merged today?

Thanks,
KT

### CODE REVIEW

File: components/authorization/app/permissions/job_permissions.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class JobPermissions
+  def self.authorized?(action, job, user)
+    WorkspacePermissions.authorized?(action, job.workspace, user)
+  end
+end
```
Comment (kevinmtrowbridge) [inline components/authorization/app/permissions/job_permissions.rb]: @AndrewLngdn: here is another permissions file that was missing from the refactoring.

**Comment by kevinmtrowbridge** [comment]
@AndrewLngdn -- @midnighteuler is running into a problem with this:

```
Started GET "/data_source_tests/629?iebuster=1460757673243" for ::ffff:127.0.0.1 at 2016-04-15 22:42:55 UTC
[e5fdf8d8-dd84-4302-ba46-ce523c100cad] [user_id:1] Processing by Api::DataSourceTestsController#show as JSON
<snip>
NameError (uninitialized constant NilClassPermissions):
```

**Comment by midnighteuler** [comment]
Here's another encountered upon browsing an HDFS data source:
```
[979d9a37-6d7f-4518-8fcd-7dcd722a30cf] [user_id:1] Completed 500 Internal Server Error in 28ms (ActiveRecord: 3.0ms)
[979d9a37-6d7f-4518-8fcd-7dcd722a30cf] [user_id:1]
NameError (uninitialized constant HdfsEntryPermissions):
  activesupport (4.1.10) lib/active_support/inflector/methods.rb:238:in `const_get'
  activesupport (4.1.10) lib/active_support/inflector/methods.rb:238:in `constantize'
  activesupport (4.1.10) lib/active_support/inflector/methods.rb:236:in `inject'
```

**Comment by kevinmtrowbridge** [comment]
I am trying to quickly get this into shape to merge in today.  Here is another Jira ticket which is stemming from this: https://alpine.atlassian.net/browse/DEV-14475

**Comment by kevinmtrowbridge** [comment]
Going back to the one that @midnighteuler reported:

```
Started GET "/data_source_tests/629?iebuster=1460757673243" for ::ffff:127.0.0.1 at 2016-04-15 22:42:55 UTC
[e5fdf8d8-dd84-4302-ba46-ce523c100cad] [user_id:1] Processing by Api::DataSourceTestsController#show as JSON
<snip>
NameError (uninitialized constant NilClassPermissions):
```

The reason this is occurring -- in the old way of doing it, [which you can see here](https://github.com/Chorus/chorus/blob/d394d1deb7d166269b41a13cece1e06026a0be9f/components/authorization_simpler/app/decorators/controllers/api/data_source_tests_controller_decorator.rb), the data_source_tests_controller_decorator.rb checks to see if the @data_source is nil, and gives permission in that special case.  This line: `return data_source_is_new(data_source) || data_sou ...`

But, this did not get brought over into the `DataSourcePermissions` object which Andrew and I created at the beginning of 6.0.  

I will add back in a test in the DataSourceTestsController ... will add back in the nil check before calling into the Authorization code:

```
def authorize
  find_test_and_datasource if @data_source.nil?
  return true if @data_source.nil?

  authorize! :update, @data_source, current_user
end
```

**Pull request merged**